### PR TITLE
should make apache run better on pi

### PR DESCRIPTION
**Pull request by jospoortvliet**
This should be simple - add some settings to httpd.conf to optimize for performance on (relatively) low-memory systems.

### CODE REVIEW

**Comment by WaaromZoMoeilijk** [comment]
I'm on this @jospoortvliet testing apache and php settings.

**Comment by oparoz** [comment]
Just make sure mpm_event is used by making it the default MPM and add a _if_ statement to those settings.

```
# MPM Event
<IfModule event.c>
    StartServers            2
    MinSpareThreads         4
    MaxSpareThreads         32
    ThreadsPerChild         16
    ServerLimit             4
    MaxRequestWorkers       64
    MaxConnectionsPerChild  0
</IfModule>
```

File: snapcraft.yaml [hunk 1]
```diff
@@ -54,6 +54,8 @@ parts:
       - authz_core
       - unixd
       - alias
+      - proxy
+      - proxy_fcgi
```
Comment (kyrofa) [inline snapcraft.yaml]: Do these actually contribute to better performance, or do they relate to another issue (and thus should be in a different PR)?

Comment (jospoortvliet) [inline snapcraft.yaml]: Oh, you're right, these are from another PR :( will remove.

File: src/owncloud/apache_config [hunk 2]
```diff
@@ -14,3 +14,18 @@ Alias "/apps" "${SNAP_DATA}/owncloud/apps"
...
+# MPM Event
+<IfModule event.c>
+    StartServers            2
+    MinSpareThreads         4
+    MaxSpareThreads         32
+    ThreadsPerChild         16
+    ServerLimit             4
+    MaxRequestWorkers       64
+    MaxConnectionsPerChild  0
+</IfModule>
```
Comment (kyrofa) [inline src/owncloud/apache_config:62]: This snap isn't runing only on the rpi. I assume these numbers would be very low for an amd64 server, for example. I'm not sure there's a good way to keep these numbers specific to the pi, but it might be worth surrounding this with:

```
<If "${SNAP_ARCH} == 'armhf'"">
...
</If>
```

or something. I'm fairly certain all the supported armhf devices would benefit from these numbers, so that seems okay to me. Need to verify that actually works, though.

File: src/owncloud/apache_config [hunk 3]
```diff
@@ -14,3 +14,19 @@ Alias "/apps" "${SNAP_DATA}/owncloud/apps"
 <IfModule dir_module>
     DirectoryIndex index.html index.php
 </IfModule>
+
+
+
+# MPM Event
+<If "${SNAP_ARCH} == 'armhf'"">
```
Comment (oparoz) [inline src/owncloud/apache_config]: Why only for arm? It's actually the default on most architectures

**Comment by oparoz** [comment]
You also need to do this

> Just make sure mpm_event is used by making it the default MPM

File: src/owncloud/apache_config (hunk 3 above)
Comment (kyrofa) [inline src/owncloud/apache_config]: @oparoz where did these numbers come from? I assumed they were targeted at the rpi, but this snap runs on more machines than the rpi. Are you saying these numbers would be good for a real amd64 server as well?

Comment (oparoz) [inline src/owncloud/apache_config]: I'm seeing from the email thread that I missed the conversation which happened during previous commits.

I would reverse the order of the if statements then.

Start with `<IfModule event.c>`, then the arch statement and we can also include one for amd64, although from my pov, it's not about the arch, but about the amount of RAM and CPUs. An amd64 VM with 1GB of RAM is not going to do much better (except it has a much better network probably)

**Comment by oparoz** [comment]
Also, I think it would be a good idea to justify each setting. The Pi is quite an unusual beast with it's puny network performance and if we don't make the config read-write, then users will have to be able to understand our decisions.

**Comment by jospoortvliet** [comment]
Interestingly, [this site](https://wiki.mikejung.biz/Apache#Event_MPM) claims you NEED php-fpm if you want to use event. Using worker now... just trying to test it.

**Comment by oparoz** [comment]
@jospoortvliet - That's absolutely right. You cannot use the non-thread safe mod_php with mm_event. I had forgotten about that...

**Comment by oparoz** [comment]
I suggest we don't waste time on workers and keep the mpm_event changes in my php-fpm branch. We can tune the config after that.

**Comment by jospoortvliet** [comment]
ok, will close this. I will then try your php-fpm branch instead, see if that works.
